File: app/src/main/java/kim/uno/util/GsonUtil.java
package kim.uno.util;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.lang.reflect.Modifier;

public class GsonUtil {

    static Gson mGson;

    public static Gson getInstance() {
        if (mGson == null)
            mGson = new GsonBuilder().excludeFieldsWithModifiers(Modifier.TRANSIENT).create();
        return mGson;
    }

    public static String toJson(Object object) {
        return getInstance().toJson(object);
    }

    public static <E> E fromJson(String json, Class<E> cls) {
        return getInstance().fromJson(json, cls);
    }
}

File: base/src/main/kotlin/kim/uno/kotlin/base/util/GsonUtil.kt
package kim.uno.kotlin.base.util

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import java.lang.reflect.Modifier

object GsonUtil {

    @JvmStatic
    val gson: Gson = GsonBuilder().excludeFieldsWithModifiers(Modifier.TRANSIENT).create()

    @JvmStatic
    fun toJson(obj: Any): String {
        return gson.toJson(obj)
    }

    @JvmStatic
    fun <E> fromJson(json: String, cls: Class<E>) : E {
        return gson.fromJson(json, cls)
    }
}